components/scream/src/share/field/field_request.hpp

```diff
@@ -233,10 +234,23 @@ struct FieldRequest {
    : FieldRequest(FID(name,layout,u,grid),std::list<std::string>{group},ps)
   { /* Nothing to do here */ }
 
+  FieldRequest (const FID& fid, const FieldRequest& parent, int idim, int k, bool dynamic)
```

Comment:
I wonder if we shouldn't rename `k` to `index` here, to emphasize that the slice index is not necessarily the vertical pack index.